### PR TITLE
ipn/localapi: remove webclient endpoint

### PR DESCRIPTION
**Pull request by soniaappasamy**
Managing starting/stopping tailscaled web client purely via setting the RunWebClient pref.

Updates tailscale/corp#14335

### CODE REVIEW

File: ipn/ipnlocal/web_client.go [hunk 1]
```diff
@@ -41,7 +40,7 @@ func (b *LocalBackend) SetWebLocalClient(lc *tailscale.LocalClient) {
 // tailscaled instance.
 // If the web interface is already running, WebClientInit is a no-op.
 func (b *LocalBackend) WebClientInit() (err error) {
-	if !hasCapability(b.netMap, tailcfg.CapabilityPreviewWebClient) {
+	if !b.ShouldRunWebClient() {
```
Comment (bradfitz) [inline ipn/ipnlocal/web_client.go:43]: (same comment as in other PR)

Comment (willnorris) [reply ipn/ipnlocal/web_client.go:43]: Do you mean the comment about whether this is just enable preview functionality?  If so, I think this one will be different.  I'm expecting that we'll leave the user pref in place, similar to what we have for SSH.  We will, however, turn it on if the user takes an explicit action to indicate they want the web client, such as running `tailscale web` in addition to the normal `tailscale set --webclient` or whatever it is (that flag doesn't exist yet).

Comment (soniaappasamy) [reply ipn/ipnlocal/web_client.go:43]: This comment https://github.com/tailscale/tailscale/pull/10086#discussion_r1380442275

I've got a change incoming

Comment (soniaappasamy) [reply ipn/ipnlocal/web_client.go:43]: Fixed in b370274b29529b308c55de108b0c6156586fc632

**Review by willnorris — APPROVED** [review]

**Comment by soniaappasamy** [comment]
Rebased off main

**Pull request merged**